client/src/lib/result.js

```diff
@@ -40,3 +47,68 @@ export function paddedAttemptResults(result, numberOfAttempts) {
   const attemptResults = result.attempts.map((attempt) => attempt.result);
   return padSkipped(attemptResults, numberOfAttempts);
 }
+
+// Events sorted by best don't need forecast view.
+// Fewest moves is currently unsupported
+// Only final rounds are supported
+export function forecastViewEnabled(round) {
+  return round.format.sortBy != "best" &&
+    round.competitionEvent.event.id != "333fm" &&
+    round.name === "Final";
+}
+
+/**
+ * return the results object with additional properties:
+ * projectedAverage: projected final average based on current results
+ * forFirst: time needed to overtake first place
+ * forThird: time needed to overtake third place
+ */
+export function resultsForView(results, format, forecastView) {
+  if (results.length == 0 || !forecastView) return results;
+  var resultsForView = results.map((result) => {
+    return {
+      ...result,
+      projectedAverage: computeProjectedAverage(result, format),
+    };
+  });
+
+  if (resultsForView[0].projectedAverage == SKIPPED_VALUE) {
+    // First place has no results. Do nothing.
+    return resultsForView;
+  }
+
+  for (let result of resultsForView) {
+    result.advancingQuestionable = false;
+  }
+
+  // Forecast view only supported for final rounds - advancing count hardcoded as 3
+  const advancingCount = 3;
+
+  // Sort based on projection with tiebreakers on single
+  resultsForView = orderBy(resultsForView, [
+    (result) => toMonotonic(result.projectedAverage),
+    (result) => toMonotonic(result.best),
+  ]);
+  resultsForView[0].ranking = 1;
+  var prevResult = resultsForView[0];
+  for (let i = 0; i < resultsForView.length; i++) {
+    let currentResult = resultsForView[i];
+    if (isSkipped(currentResult.projectedAverage)) {
+      break;
+    }
+    if (currentResult.projectedAverage === prevResult.projectedAverage &&
+      currentResult.best === prevResult.best) {
```

Comment:
For comparison we should use `toMonotonic`, so that DNF DNS is treated the same:

```suggestion
    if (toMonotonic(currentResult.projectedAverage) === toMonotonic(prevResult.projectedAverage) &&
      toMonotonic(currentResult.best) === toMonotonic(prevResult.best)) {
```